snap/squashfs/squashfs.go

```diff
@@ -117,6 +118,99 @@ func (s *Snap) ReadFile(filePath string) (content []byte, err error) {
 	return ioutil.ReadFile(filepath.Join(unpackDir, filePath))
 }
 
+// skippper could also be a map[string]bool, but _as it's only supposed to be
+// checkd through its Has method_, the small added complexity of it being a
+// map[string]struct{} lose to the associated space savings.
+type skipper map[string]struct{}
+
+func (sk skipper) Add(path string) {
+	sk[filepath.Clean(path)] = struct{}{}
+}
+
+func (sk skipper) Has(path string) bool {
+	for p := filepath.Clean(path); p != "." && p != "/"; p = filepath.Dir(p) {
+		if _, ok := sk[p]; ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Stat returns an os.FileInfo of a single file or directory inside a squashfs snap.
```

Comment:
Maybe expand the comment a little bit to point out that the os.FileInfo referenced here referes to the arguments that filepath.WalkerFunc receives?

Reply:
uh, that comment is broken. Fixing.